### PR TITLE
HUD revisions

### PR DESCRIPTION
**Pull request by Sezzary**
Introduces a fancy OpenLara-inspired pickup summary.
- Pickups will travel from a projected 2D point on the screen to a stack on the right.
- The stack has a max height of 6. If filled in the uncommon case, new pickups are stashed offscreen above.
- Pickups of the same type are grouped together with a count denoted by a numeric string.
- Each pickup has a limited display life. The stack funnels through one passage as display lives end.

Additional changes:
- Added `Hud` folder containing `Hud.cpp/h` and `PickupSummary.cpp/h`.
- Wrapped new classes and old health functions under new `TEN::Hud` namespace.
- Added the beginnings of a simple `HudController` class.
- Added rudimentary ability to scale strings.
- Other minor changes as I passed through functionality that was needed for the pickup summary.

### CODE REVIEW

File: TombEngine/Game/Hud/PickupSummary.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+				return;
+			}
+		}
+
+		// Create new display pickup.
+		auto& pickup = GetNewDisplayPickup();
+
+		pickup.ObjectID = objectID;
+		pickup.Count = 1;
+		pickup.Position =
```
Comment (Kubsy) [inline TombEngine/Game/Hud/PickupSummary.cpp:122]: oopsie

**Comment by Kubsy** [comment]
https://user-images.githubusercontent.com/80340234/216782066-bbd36563-2708-4c25-a796-b1cfba979aad.mp4

pickups do not display

File: TombEngine/Game/Hud/PickupSummary.cpp (hunk 1 above)
Comment (Sezzary) [reply TombEngine/Game/Hud/PickupSummary.cpp:122]: Nope, this is intentional. :P `Position` is assigned to the line below.

Comment (Kubsy) [reply TombEngine/Game/Hud/PickupSummary.cpp:122]: oh okay, first time seeing this aha

**Comment by Kubsy** [comment]
just to note: the revolver appears with lasersight in the inventory already attached but on other branch it appears without the lasersight.

**Comment by Stranger1992** [comment]
**Tests**

Pickups displaying correctly - OCB 0 ✅

**Review by Adngel — COMMENTED** [review]
I like the direction where this is going, although I'm afraid I'm not very sure how to test most of the HUD functions.

In any case, I had a problem compiling the branch that I think it should be fixed. But apart of that I haven't got any objection with this branch.

File: TombEngine/Renderer/Renderer11Helper.cpp [hunk 2]
```diff
@@ -532,4 +505,65 @@ namespace TEN::Renderer
...
\ No newline at end of file
+
+	Vector2i Renderer11::GetScreenResolution() const
+	{
+		return Vector2i(m_screenWidth, m_screenHeight);
+	}
+
+	Vector2 Renderer11::GetScreenSpacePosition(const Vector3& pos) const
+	{
+		constexpr auto INVALID_SCREEN_SPACE_POSITION = Vector2(INT_MAX, INT_MAX);
```
[the remark this answers is not in the document]
Comment (Sezzary) [reply TombEngine/Renderer/Renderer11Helper.cpp:516]: Sad result of a quick commit. Will fix.

File: TombEngine/Game/Lara/lara_objects.cpp [hunk 3]
```diff
@@ -35,9 +35,6 @@ void lara_as_pickup(ItemInfo* item, CollisionInfo* coll)
 	lara->Control.CanLook = false;
 	coll->Setup.EnableObjectPush = false;
 	coll->Setup.EnableSpasm = false;
-	Camera.targetAngle = -ANGLE(130.0f);
-	Camera.targetElevation = -ANGLE(15.0f);
-	Camera.targetDistance = SECTOR(1);
```
[the remark this answers is not in the document]
Comment (Sezzary) [reply TombEngine/Game/Lara/lara_objects.cpp:40]: This was done for testing purposes, just forgot about it. Will revert.

**Review by Adngel — APPROVED** [review]

**Review by Kubsy — APPROVED** [review]
everything works as expected.

**Review by Adngel — APPROVED** [review]

**Pull request merged**